### PR TITLE
initial draft of integration tests for polarion

### PR DESCRIPTION
**Pull request by jscotka**

### CODE REVIEW

File: tests/integration/data/polarion/existing_testcase/main.fmf [hunk 1]
```diff
@@ -3,4 +3,5 @@ extra-nitrate: TC#0609686
 link:
   - verifies: https://bugzilla.redhat.com/show_bug.cgi?id=1925518
   - implements: https://polarion.example/polarion/#/project/TMT/workitem?id=TMT-43
+  - implements: https://polarion.engineering.redhat.com/polarion/#/project/RedHatEnterpriseLinux7/workitem?id=RHEL-293393
```
Comment (KwisatzHaderach) [inline tests/integration/data/polarion/existing_testcase/main.fmf:6]: either make it .example like ^ or IMO you can remove this line completely so we are not putting internal links on public github

right now it doesn't use the links for anything anyway

**Review by KwisatzHaderach — APPROVED** [review]
Good job getting requre to work with pylero.

**Review by lukaszachy — CHANGES_REQUESTED** [review]
LGTM but please rebase to the current main.

**Comment by psss** [comment]
Hm, I'm not really sure about the binary data. The `PolarionBase.test.yaml` file has over 2 MB. @jscotka, if there is a slight change of the call, would the large blob be regenerated or not? If the expected Polarion data are expected to be in megabytes I'd strongly vote for a separate repository for these integration tests. Perhaps we could create a https://github.com/teemtee/tests repo for extra test coverage which is not suitable for the main repo?

**Comment by thrix** [comment]
> Hm, I'm not really sure about the binary data. The `PolarionBase.test.yaml` file has over 2 MB. @jscotka, if there is a slight change of the call, would the large blob be regenerated or not? If the expected Polarion data are expected to be in megabytes I'd strongly vote for a separate repository for these integration tests. Perhaps we could create a https://github.com/teemtee/tests repo for extra test coverage which is not suitable for the main repo?

Another idea: couldn't we compress this data, from 2MB we get 200kB with xz, what is definitely better?

**Comment by thrix** [comment]
> > Hm, I'm not really sure about the binary data. The `PolarionBase.test.yaml` file has over 2 MB. @jscotka, if there is a slight change of the call, would the large blob be regenerated or not? If the expected Polarion data are expected to be in megabytes I'd strongly vote for a separate repository for these integration tests. Perhaps we could create a https://github.com/teemtee/tests repo for extra test coverage which is not suitable for the main repo?
> 
> Another idea: couldn't we compress this data, from 2MB we get 200kB with xz, what is definitely better?

And second idea, what Polarion objects did we use, couldn't we spare some space also using some different data?

**Comment by lukaszachy** [comment]
Another way is to use git lfs to store those files - https://docs.github.com/en/repositories/working-with-files/managing-large-files/configuring-git-large-file-storage

**Comment by jscotka** [comment]
> Hm, I'm not really sure about the binary data. The `PolarionBase.test.yaml` file has over 2 MB. @jscotka, if there is a slight change of the call, would the large blob be regenerated or not? If the expected Polarion data are expected to be in megabytes I'd strongly vote for a separate repository for these integration tests. Perhaps we could create a https://github.com/teemtee/tests repo for extra test coverage which is not suitable for the main repo?

another repo is also possible, but probbaly not ideal case. I think there shouldnt ne much changes often. but I understand it is more data than expected. Especially it is exactly case, where when TMT sync data into/back test machines causes issues, because it is bigger than obvious.

**Comment by jscotka** [comment]
> > Hm, I'm not really sure about the binary data. The `PolarionBase.test.yaml` file has over 2 MB. @jscotka, if there is a slight change of the call, would the large blob be regenerated or not? If the expected Polarion data are expected to be in megabytes I'd strongly vote for a separate repository for these integration tests. Perhaps we could create a https://github.com/teemtee/tests repo for extra test coverage which is not suitable for the main repo?
> 
> Another idea: couldn't we compress this data, from 2MB we get 200kB with xz, what is definitely better?

yep, possible, but from perspective of data, have it compressed means that it will changes everything when changed, without compression there is change that there will be just some diff.

**Comment by jscotka** [comment]
> > > Hm, I'm not really sure about the binary data. The `PolarionBase.test.yaml` file has over 2 MB. @jscotka, if there is a slight change of the call, would the large blob be regenerated or not? If the expected Polarion data are expected to be in megabytes I'd strongly vote for a separate repository for these integration tests. Perhaps we could create a https://github.com/teemtee/tests repo for extra test coverage which is not suitable for the main repo?
> > 
> > 
> > Another idea: couldn't we compress this data, from 2MB we get 200kB with xz, what is definitely better?
> 
> And second idea, what Polarion objects did we use, couldn't we spare some space also using some different smaller data?

It is also possible, that we can use these requre data analyse and try to improve polarion usage, what also will lead to higher speed or less data usage of TMT itself. But these optimization are probably not current goal.

**Comment by jscotka** [comment]
> 

Hi @thrix finally I've done smaller hack to store binary data as separate files and compress them to one archive, so that core data are clear and binary data are also easier to look at and see what is there, just need to decompress them before tests.  What is part of TMT polarion testplan. It seems clear and it is <100kb of data. So I think it is now nice solution.

And you are right theoretically I can filter objects form polarion to just these what we really use, but it could be very hard to do. theoretically as post-processing these data from polarion.

File: plans/integration/create_pylero_cfg.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+FILE=~/.pylero
+
+if [ ! -f $FILE ]; then
+
+echo "[webservice]
+url=https://polarion.example.com/polarion
+svn_repo=https://polarion.examplee.com/repo
```
Comment (KwisatzHaderach) [inline plans/integration/create_pylero_cfg.sh:8]: not like it matters, but you have a typo here 'examplee'

File: tests/integration/data/polarion/existing_dryrun_testcase/main.fmf [hunk 3]
```diff
@@ -3,4 +3,5 @@ extra-nitrate: TC#0609686
 link:
   - verifies: https://bugzilla.redhat.com/show_bug.cgi?id=1925518
   - implements: https://polarion.example/polarion/#/project/TMT/workitem?id=TMT-43
-id: 407a160f-86a8-49b0-aff6-634fc1aa2890
+  - implements: https://polarion.engineering.redhat.com/polarion/#/project/RHELBASEOS/workitem?id=BASEOS-12730
```
Comment (KwisatzHaderach) [inline tests/integration/data/polarion/existing_dryrun_testcase/main.fmf:6]: this line should probably not be here, internal link

**Review by KwisatzHaderach — APPROVED** [review]

**Review by KwisatzHaderach — APPROVED** [review]

**Comment by psss** [comment]
Thanks for minimizing the test data! After a short discussion with @jscotka we agreed to give a try to a dedicated tests repository. This will serve as a

 * space for tests which need large or binary data
 * dogfooding the fresh remote plan import feature

The repository has been created here:

 * https://github.com/teemtee/tests

Packit is enabled so we can enable testing of the changes in the repo itself as well.

**Comment by jscotka** [comment]
> Thanks for minimizing the test data! After a short discussion with @jscotka we agreed to give a try to a dedicated tests repository. This will serve as a
> 
>     * space for tests which need large or binary data
> 
>     * dogfooding the fresh remote plan import feature
> 
> 
> The repository has been created here:
> 
>     * https://github.com/teemtee/tests
> 
> 
> Packit is enabled so we can enable testing of the changes in the repo itself as well.

So, moved tests to https://github.com/teemtee/tests/pull/1

**Comment by jscotka** [comment]
closing this PR, as we use remote plan for polarion tests

**Pull request closed without merging**